packages/common/src/discussions/utils.ts

```diff
@@ -43,13 +49,101 @@ export function setDiscussableKeyword(
   return updatedTypeKeywords;
 }
 
+/**
+ * Determines if the given channel is considered to be a `public` channel, supporting both
+ * legacy permissions and V2 ACL model.
+ * @param channel An IChannel record
+ * @returns true if the channel is considered `public`
+ */
+export function isPublicChannel(channel: IChannel): boolean {
+  return channel.channelAcl
+    ? channel.channelAcl.some(
+        ({ category }) => category === AclCategory.AUTHENTICATED_USER
+      )
+    : channel.access === SharingAccess.PUBLIC;
+}
+
+/**
+ * Determines if the given channel is considered to be an `org` channel, supporting both
+ * legacy permissions and V2 ACL model.
+ * @param channel An IChannel record
+ * @returns true if the channel is considered `org`
+ */
+export function isOrgChannel(channel: IChannel): boolean {
+  return channel.channelAcl
+    ? channel.channelAcl.some(({ category }) => category === AclCategory.ORG)
```

Comment:
This should also check for `subCategory` of `MEMBER`, since public channels will have category: org, subCategory: admin set to `owner`, which would cause this to return incorrectly.

And if this function is available to be used without the other helpers, should also verify it's not a public channel
```js
channel.channelAcl.some(({ category, subCategory }) => category === AclCategory.ORG && subCategory === AclSubCategory.MEMBER) && !isPublicChannel(channel)
```